src/Component/Tests/Symfony/Maker/MakeResourceTest.php

```diff
@@ -0,0 +1,337 @@
+
+    private const EXISTING_DUMMY_REPOSITORY_PATH = 'src/Repository/Entity/ExistingDummyRepository.php';
+
+    /** @test */
+    public function it_can_create_resources(): void
+    {
+        $tester = new CommandTester((new Application(self::bootKernel()))->find('make:resource'));
+
+        $this->assertFileDoesNotExist(self::file(self::BOOK_RESOURCE_PATH));
+
+        $tester->execute(['name' => '\\App\\Tests\\Tmp\\Sylius\\Resource\\BookResource', '--is-entity' => false]);
+
+        $this->assertFileExists(self::file(self::BOOK_RESOURCE_PATH));
+        $this->assertSame(self::getBookResourceExpectedContent(), \file_get_contents(self::file(self::BOOK_RESOURCE_PATH)));
+    }
+
+    /** @test */
+    public function it_can_create_resource_entities(): void
+    {
+        $tester = new CommandTester((new Application(self::bootKernel()))->find('make:resource'));
+
+        $this->assertFileDoesNotExist(self::file(self::DUMMY_RESOURCE_PATH));
+        $this->assertFileDoesNotExist(self::file(self::DUMMY_REPOSITORY_PATH));
+
+        $tester->execute(['name' => '\\App\\Tests\\Tmp\\Entity\\Dummy', '--is-entity' => true]);
+
+        $this->assertFileExists(self::file(self::DUMMY_RESOURCE_PATH));
+        $this->assertSame(self::getDummyResourceExpectedContent(), \file_get_contents(self::file(self::DUMMY_RESOURCE_PATH)));
+
+        $this->assertFileExists(self::file(self::DUMMY_REPOSITORY_PATH));
+        $this->assertSame(self::getDummyRepositoryExpectedContent(), \file_get_contents(self::file(self::DUMMY_REPOSITORY_PATH)));
+    }
+
+    /** @test */
+    public function it_can_update_entities(): void
+    {
+        $this->assertFileDoesNotExist(self::file(self::EXISTING_DUMMY_RESOURCE_PATH));
+        $this->assertFileDoesNotExist(self::file(self::EXISTING_DUMMY_REPOSITORY_PATH));
+
+        $tester = new CommandTester((new Application(self::bootKernel()))->find('make:entity'));
+        $tester->setInputs(['name', 'string', 255, 'y', '']);
+        $tester->execute(['name' => '\\App\\Tests\\Tmp\\Entity\\ExistingDummy']);
+
+        $this->assertFileExists(self::file(self::EXISTING_DUMMY_RESOURCE_PATH));
+        $this->assertFileExists(self::file(self::EXISTING_DUMMY_REPOSITORY_PATH));
+
+        $tester = new CommandTester((new Application(self::bootKernel()))->find('make:resource'));
+        $tester->execute(['name' => '\\App\\Tests\\Tmp\\Entity\\ExistingDummy', '--is-entity' => true]);
+
+        $this->assertSame(self::getUpdatedDummyResourceExpectedContent(), \file_get_contents(self::file(self::EXISTING_DUMMY_RESOURCE_PATH)));
+        $this->assertSame(self::getUpdatedDummyRepositoryExpectedContent(), \file_get_contents(self::file(self::EXISTING_DUMMY_REPOSITORY_PATH)));
+    }
+
+    private static function getBookResourceExpectedContent(): string
+    {
+        return <<<EOF
+            <?php
+            
+            namespace App\Tests\Tmp\Sylius\Resource;
```

Comment:
Why do we have `Sylius` in the namespace for resources but not for processors? We have the command `make:resource` that have Sylius in a namespace, but then we have `make:sylius-state-processor` command, which generates processor without Sylius in namespace.